GVFS/GVFS/CommandLine/DehydrateVerb.cs

```diff
@@ -145,31 +164,139 @@ protected override void Execute(GVFSEnlistment enlistment)
                 // Local cache and objects paths are required for TryDownloadGitObjects
                 this.InitializeLocalCacheAndObjectsPaths(tracer, enlistment, retryConfig, serverGVFSConfig: null, cacheServer: null);
 
-                if (this.TryBackupFiles(tracer, enlistment, backupRoot))
+                if (string.IsNullOrEmpty(this.Folders))
                 {
-                    if (this.TryDownloadGitObjects(tracer, enlistment, retryConfig) &&
-                        this.TryRecreateIndex(tracer, enlistment))
+                    this.RunFullDehydrate(tracer, enlistment, backupRoot, retryConfig);
+                }
+                else
+                {
+                    string[] folders = this.Folders.Split(new[] { ";" }, StringSplitOptions.RemoveEmptyEntries);
+
+                    if (folders.Length > 0)
                     {
-                        // Converting the src folder to partial must be the final step before mount
-                        this.PrepareSrcFolder(tracer, enlistment);
-                        this.Mount(tracer);
+                        if (cleanStatus)

```

Comment:
The output for `CheckGitStatus` says:

> "If this takes too long, you can abort and run dehydrate with --no-status to skip this safety check."

But it looks like that is not true when a user specifies specific folders.  Omit that line or update the text to indicate that it can only be used in "full dehydrate" mode (assuming we always want to require it)?  (Same comment for that option's `HelpText`).